src/ParquetMsg.chpl

```diff
@@ -705,6 +725,102 @@ module ParquetMsg {
     }
   }
 
+  inline proc getReaderIdx(fileNum: int, rgNum: int) {
+    // we can assume there won't be more than 1000 RGs in a file
+    return (fileNum*1000) + rgNum;
+  }
+
+  proc getRowGroupNums(ref distFiles, ref numRowGroups) {
+    coforall loc in distFiles.targetLocales() with (ref numRowGroups) do on loc {
+      var locFiles: [distFiles.localSubdomain()] string = distFiles[distFiles.localSubdomain()];
+      for i in locFiles.domain {
+        c_openFile(locFiles[i].localize().c_str(), getReaderIdx(i,0));
+        numRowGroups[i] = c_getNumRowGroups(getReaderIdx(i,0));
+        for j in 2..numRowGroups[i] {
+          c_openFile(locFiles[i].localize().c_str(), getReaderIdx(i,j-1));
+        }
```

Comment:
I don't totally get why `j` starts at `2` and not `1` because we have `getReaderIdx(i,0)` calls ahead of this

ohhh I think I just answered my own question lol. the first bit handles the first row group in the file and the for loop handles any remaining row groups...

which leads me to the question, why not remove the first 2 lines and just have 
```chapel
for 1 in 2..numRowGroups[i] {
  c_openFile(locFiles[i].localize().c_str(), getReaderIdx(i,j-1));
}
```

Reply:
Ya, good point.

Reply:
Actually, we need this, the loop requires that we know the number of row groups, which we don't know until the first one has been computed.

Reply:
ahhhh okay! that makes sense